src/core/layout/qgslayoutexporter.cpp

```diff
@@ -346,7 +346,13 @@ QgsLayoutExporter::ExportResult QgsLayoutExporter::exportToImage( const QString
   PageExportDetails pageDetails;
   pageDetails.directory = fi.path();
   pageDetails.baseName = fi.baseName();
-  pageDetails.extension = fi.completeSuffix();
+  pageDetails.extension = fi.suffix();
+  // Deal with multiple dots, issue GH-29980
+  const auto completeSuffix { fi.completeSuffix() };
+  if ( completeSuffix != pageDetails.extension )
```

Comment:
Would completeBaseName work here instead?

Reply:
Yes, sure, I missed that one.